### DISCUSSION TITLE
How to scroll to a selector with a button in the sidebar?

### DISCUSSION THREAD

**Discussion by BurningTreeC** [TiddlyWiki/TiddlyWiki5#6631] [Q&A] [answered]
Hi all,

The sidebar is wrapped into a scrollable widget which intercepts the tm-scroll message

But I'd like to scroll to a domNode which is in the river using a button in the sidebar that dispatches the tm-scroll message with a selector

Is there any possibility doing this or any chance to apply a workaround to the core?

**Comment by Jermolene** [comment, marked as answer]
Ah, good point. The scrollable widget should not trap `tm-scroll` messages if they include a `selector` parameter.

**Reply by BurningTreeC** [reply]
Thanks @Jermolene , should I prepare a PR?

**Reply by Jermolene** [reply]
I think that would make sense, yes, thank you!

**Reply by BurningTreeC** [reply]
@Jermolene , @saqimtiaz mentioned that it wouldn't be backwards compatible if we do so
He's in favor of adding a switch like `passThrough="yes"`

**Reply by saqimtiaz** [reply]
To be more specific this is what I stated:

> If a tm-scroll message arises inside a scrollable widget and the selector targets a DOM element inside the scrollable, will the scrolling still happen inside the scrollable and not at the page scroller level?

 > Some sort of passthrough parameter is indeed a possibility. I think we need to consider this holistically along with the changes to scrolling proposed in https://github.com/Jermolene/TiddlyWiki5/pull/6602. Changes to how we handle scrolling could in fact make a case for deprecating the scrollable widget.

**Reply by saqimtiaz** [reply]
See https://github.com/Jermolene/TiddlyWiki5/pull/6643#issuecomment-1100802125

**Reply by Jermolene** [reply]
> @Jermolene , @saqimtiaz mentioned that it wouldn't be backwards compatible if we do so
> He's in favor of adding a switch like `passThrough="yes"`

I think all that's needed is for the handler in the scrollable widget to pass the message through if the `selector` parameter is present.

**Reply by saqimtiaz** [reply]
> I think all that's needed is for the handler in the scrollable widget to pass the message through if the `selector` parameter is present.

I do not believe that would be backwards compatible @Jermolene.

Consider the following scenario:

- a `tm-sroll` message with a `selector` parameter originates from within a scrollable
- the `selector` targets a DOM element inside the same scrollable
- the `tm-scroll` message is handled by the `scrollable` widget and scrolling happens within the scrollable to bring the target element in to view. No scrolling happens at the document level. This is the current behaviour.

If we allow `tm-scroll` messages with a `selector` parameter to passthrough the $scrollable widget, the scrolling will be handled at the page/document level and no scrolling will occur inside the `scrollable`. Unless I am missing something, this will both be inconsistent with current behaviour where no scrolling happens at the document level, and it will also remove the only affordance for programmatically triggering scrolling inside a `$scrollable` widget from wikitext. This would be a breaking change for several of my wikis.

**Reply by Jermolene** [reply]
Yes of course, you're quite right. I'd forgotten we'd made the scrollable widget aware of the selector parameter, as well as the page scroller.

Could we pass the tm-scroll through if there is a selector parameter, and also the selector identifies an element that is not a descendent of the scrollable widget?

**Reply by saqimtiaz** [reply]
> Could we pass the tm-scroll through if there is a selector parameter, and also the selector identifies an element that is not a descendent of the scrollable widget?

I'll paraphrase that as allowing the `tm-scroll` through if there is a `selector` parameter and no matching element that is a descendant of the scrollable widget, as that seems closer to the current behaviour. Otherwise users would need to ensure that their selectors are unique not only in the scrollable but document wide, breaking current behaviour.

Strictly speaking that would not be backwards compatible either, though not a problem for my particular use case.

Currently you can send a `tm-scroll` message with a `selector` inside a scrollable widget in the knowledge that if there is no matching target element in the scrollable, no scrolling will happen. If we then allow that message to propagate, an element outside the scrollable might match and trigger scrolling whereas with the current code that would not happen. My main worry here is that it would remove the affordance of "scroll to this selector if the element exists in the scrollable, otherwise do nothing".
